sycl/include/sycl/ext/intel/experimental/esimd/math.hpp

```diff
@@ -1727,6 +1727,91 @@ __ESIMD_API __ESIMD_NS::simd<T, N> dpasw2(
 }
 /// @} sycl_esimd_systolic_array_api
 
+/// @addtogroup sycl_esimd_logical
+/// @{
+
+/// Performs binary function computation with three vector operands.
+/// @tparam FuncControl boolean function control expressed with bfn_t
+/// enum values. Encodes all possible 256 functions with three boolean
+/// operands in an unsigned char.
+/// @tparam T type of the input vector element.
+/// @tparam N size of the input vector.
+/// @param s0 First boolean function argument.
+/// @param s1 Second boolean function argument.
+/// @param s2 Third boolean function argument.
+///
+/// Example: d = bfn<~bfn_x & ~bfn_y & ~bfn_z>(s0, s1, s2);
+
+enum class bfn_t : uint8_t { x = 0xAA, y = 0xCC, z = 0xF0 };
+
+/// The first argument of the boolean function.
+constexpr bfn_t bfn_x = bfn_t::x;
```

Comment:
is it possible to put all the bfn_t stuff inside a stuct so it's encapsulated? or does it have to be seperate?

also is it possible to move it to some detail header?

Reply:
IMO, these bfn_x, bfn_y, bfn_z are excessive. Having bfn_t is more than enough.

Reply:
> is it possible to put all the bfn_t stuff inside a stuct so it's encapsulated? or does it have to be seperate?
> 
> also is it possible to move it to some detail header?

I want to continue having bfn_t castable to uint8_t. Or you had something else in mind?
I looked at experimental detail header for math, it looks like it mostly contains `__esimd_*` intrinsics, I'm not sure it would belong there. Let me know what you think.

Reply:
> IMO, these bfn_x, bfn_y, bfn_z are excessive. Having bfn_t is more than enough.

Removed.